core/src/main/java/nl/nn/adapterframework/receivers/JavaListener.java

```diff
@@ -57,7 +57,7 @@
  *
  * @author  Gerrit van Brakel
  */
-public class JavaListener implements IPushingListener<String>, RequestProcessor, HasPhysicalDestination {
+public class JavaListener implements IPushingListener<Message>, RequestProcessor, HasPhysicalDestination {
```

Comment:
This is or a different approach is necessary since we store Message in database and try to extract data while using message filter in error store. Currently, it throws ClassCastException since the extractMessage expects string when we provide Message. Another approach to solve this could be to pass string instead of message to a javaListener extractMessage method. But I think it is better to start using Message type in JavaListener

Reply:
But currently JavaListener is of type IPushingListener<String>, and should work like that. This change appears to me as a workaround for another problem.
I think the problem is in MessageBrowsingFilter.getRawMessage().
1) It does not return a rawMessage, but returns messagebody-as-text. This method should be called something like getMessageText 
2) on line 176, the result of extractMessage, that is a Message, is assigned to a String.

This problem becomes clear when the signature of getRawMessage is specified better, like:
`private String getRawMessage(IMessageBrowser<?> messageBrowser, IListener<?> listener, String messageId)